account_fiscal_year/models/date_range.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Author: Damien Crier
+# Copyright 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import models, api, _, exceptions
+
+
+class DateRange(models.Model):
+    _inherit = 'date.range'
+
+    @api.multi
+    def unlink(self):
+        """
+        Cannot delete a date_range of type 'fiscal_year'
+        """
```

Comment:
I would *not* forbid deletion of date.range. Should that become necessary on some specific contexts it can be done with a record rule.

On the contrary, preventing deletion of date.range.type is necessary for preserving correct behaviour of the module.